pytools/graph.py

```diff
@@ -316,9 +355,17 @@ def compute_topological_order(graph: GraphT,
                 heappush(heap, HeapEntry(child, keyfunc(child)))
 
     if len(order) != total_num_nodes:
-        # any node which has a predecessor left is a part of a cycle
-        raise CycleError(next(iter(n for n, num_preds in
-            nodes_to_num_predecessors.items() if num_preds != 0)))
+        # There is a cycle in the graph
+        if not verbose_cycle:
+            raise CycleError(None)
+
+        try:
+            cycles = find_cycles(graph)
+        except KeyError:
+            # Graph is invalid
+            raise CycleError(None)
+        else:
+            raise CycleError(cycles[0][0])
```

Comment:
Add to the documentation of `CycleError` what the value might mean.

Reply:
Hmm, the current doc of `CycleError` has `:attr node: Node in a directed graph that is part of a cycle.` - I'm not sure what else to add there.